ansible_runner/streaming.py

```diff
@@ -0,0 +1,227 @@
+        self.status = status_data['status']
+        self._output.write(json.dumps(status_data).encode('utf-8'))
+        self._output.write(b'\n')
+        self._output.flush()
+
+    def event_handler(self, event_data):
+        self._output.write(json.dumps(event_data).encode('utf-8'))
+        self._output.write(b'\n')
+        self._output.flush()
+
+    def artifacts_handler(self, artifact_dir):
+        self._output.write(utils.stream_dir(artifact_dir))
+        self._output.flush()
+
+    def finished_callback(self, runner_obj):
+        self._output.write(json.dumps({'eof': True}).encode('utf-8'))
+        self._output.write(b'\n')
+        self._output.flush()
+
+
+class Processor(object):
+    def __init__(self, _input=None, status_handler=None, event_handler=None,
+                 artifacts_handler=None, cancel_callback=None, finished_callback=None, **kwargs):
+        if _input is None:
+            _input = sys.stdin.buffer
+        self._input = _input
+
+        private_data_dir = kwargs.get('private_data_dir')
+        if private_data_dir is None:
+            private_data_dir = tempfile.TemporaryDirectory().name
+        self.private_data_dir = private_data_dir
+        self._loader = ArtifactLoader(self.private_data_dir)
+
+        settings = kwargs.get('settings')
+        if settings is None:
+            try:
+                settings = self._loader.load_file('env/settings', Mapping)
+            except ConfigurationError:
+                settings = {}
+        self.config = MockConfig(settings)
+
+        artifact_dir = kwargs.get('artifact_dir')
+        self.artifact_dir = os.path.abspath(
+            artifact_dir or os.path.join(self.private_data_dir, 'artifacts'))
+
+        self.status_handler = status_handler
+        self.event_handler = event_handler
+        self.artifacts_handler = artifacts_handler
+
+        self.cancel_callback = cancel_callback  # FIXME: unused
+        self.finished_callback = finished_callback
+
+        self.status = "unstarted"
+        self.rc = None
+
+    def status_callback(self, status_data):
+        self.status = status_data['status']
+        if self.status == 'starting':
+            self.config.command = status_data.get('command')
+            self.config.env = status_data.get('env')
+            self.config.cwd = status_data.get('cwd')
+
+        for plugin in ansible_runner.plugins:
+            ansible_runner.plugins[plugin].status_handler(self.config, status_data)
+        if self.status_handler is not None:
+            self.status_handler(status_data, runner_config=self.config)
+
+    def event_callback(self, event_data):
+        full_filename = os.path.join(self.artifact_dir,
+                                     'job_events',
+                                     '{}-{}.json'.format(event_data['counter'],
+                                                         event_data['uuid']))
+        if 'stdout' in event_data:
+            print(event_data['stdout'])
```

Comment:
Since `process` is taking in kwargs and private_data_dir, you could just assign the `quiet` config to an instance variable, and check that to determine if this should/shouldn't be printed. And the worker could always set quiet=True.